Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/ComponentEditor.cpp

```diff
@@ -924,19 +924,22 @@ namespace AzToolsFramework
 
     void ComponentEditor::EnteredComponentMode(const AZStd::vector<AZ::Uuid>& componentModeTypes)
     {
+        if (componentModeTypes.empty())
+        {
+            SetWidgetInteractEnabled(this, false);
+        }
+
         // disable all component cards not matching the ComponentMode type
-        if (AZStd::find(
-            componentModeTypes.begin(),
-            componentModeTypes.end(), m_componentType) == componentModeTypes.end())
+        if (AZStd::find(componentModeTypes.begin(), componentModeTypes.end(), m_componentType) == componentModeTypes.end())
         {
             SetWidgetInteractEnabled(this, false);
         }
         else
         {
-            if (!componentModeTypes.empty())
+            if (componentModeTypes.front() == m_componentType)
             {
                 // only set the first item to be selected/highlighted
-                SetSelected(componentModeTypes.front() == m_componentType);
+                SetSelected(true);
```

Comment:
TLDR: I think all of this can get reverted?

Longer version:

This part is redundant, because the find() call will _also_ set things to false if componentModeTypes is empty, since the find() would always return false.
```
       if (componentModeTypes.empty())
        {
            SetWidgetInteractEnabled(this, false);
        }
```

This part I think breaks the logic in here, because now there's nothing calling SetSelected(false) in the case where componentModeTypes is NOT empty and the widget is in componentModeTypes and the widget isn't the first one:
```
            if (componentModeTypes.front() == m_componentType)
            {
                // only set the first item to be selected/highlighted
                SetSelected(true);
```

My suggestion would actually be to shorten the implementation down a little bit to this:
```
        if (AZStd::find(
            componentModeTypes.begin(),
            componentModeTypes.end(), m_componentType) == componentModeTypes.end())
        {
            // disable all component cards that aren't in the active component mode.
            SetWidgetInteractEnabled(this, false);
        }
        else
        {
            // For components that *are* in the active component mode, try to set the first one
            // to selected, and the rest to unselected.
            SetSelected(componentModeTypes.front() == m_componentType);
        }
```